src/catalog/catalog.cpp

```diff
@@ -257,24 +147,22 @@ void CatalogContent::readFromFile(const std::string& directory, DBFileType dbFil
     auto fileInfo = FileUtils::openFile(catalogPath, O_RDONLY);
     uint64_t offset = 0;
     validateMagicBytes(fileInfo.get(), offset);
-    storage::storage_version_t savedStorageVersion;
+    storage_version_t savedStorageVersion;
     SerDeser::deserializeValue<uint64_t>(savedStorageVersion, fileInfo.get(), offset);
     validateStorageVersion(savedStorageVersion);
     uint64_t numNodeTables, numRelTables;
     SerDeser::deserializeValue<uint64_t>(numNodeTables, fileInfo.get(), offset);
     SerDeser::deserializeValue<uint64_t>(numRelTables, fileInfo.get(), offset);
     table_id_t tableID;
     for (auto i = 0u; i < numNodeTables; i++) {
-        SerDeser::deserializeValue<uint64_t>(tableID, fileInfo.get(), offset);
-        nodeTableSchemas[tableID] = std::make_unique<NodeTableSchema>();
-        SerDeser::deserializeValue<NodeTableSchema>(
-            *nodeTableSchemas[tableID], fileInfo.get(), offset);
+        SerDeser::deserializeValue<table_id_t>(tableID, fileInfo.get(), offset);
+        nodeTableSchemas[tableID] = ku_static_unique_pointer_cast<TableSchema, NodeTableSchema>(
```

Comment:
I think you don't need to do a pointer cast, c++ will do it automatically for you

Reply:
This is a downward casting, which cannot be handled automatically by cpp.